src/Components/WebAssembly/WebAssembly/src/HotReload/HotReloadAgent.cs

```diff
@@ -4,227 +4,131 @@
 // Based on the implementation in https://raw.githubusercontent.com/dotnet/sdk/aad0424c0bfaa60c8bd136a92fd131e53d14561a/src/BuiltInTools/DotNetDeltaApplier/HotReloadAgent.cs
 
 using System.Collections.Concurrent;
+using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
 using System.Linq;
 using System.Reflection;
-using System.Reflection.Metadata;
 
 namespace Microsoft.Extensions.HotReload;
 
+[UnconditionalSuppressMessage("Trimming", "IL2026", Justification = "Hot reload is only expected to work when trimming is disabled.")]
 internal sealed class HotReloadAgent : IDisposable
 {
-    /// Flags for hot reload handler Types like MVC's HotReloadService.
-    private const DynamicallyAccessedMemberTypes HotReloadHandlerLinkerFlags = DynamicallyAccessedMemberTypes.PublicMethods | DynamicallyAccessedMemberTypes.NonPublicMethods;
+    private const string MetadataUpdaterTypeName = "System.Reflection.Metadata.MetadataUpdater";
```

Comment:
Copied from dotnet-watch. Will be shared.

Reply:
> Will be shared

Is this going to be done in a follow-up somewhere?

Reply:
Yes. Working on it here: https://github.com/dotnet/sdk/pull/44899